src/contact.rs

```diff
@@ -27,6 +29,67 @@ use crate::param::{Param, Params};
 use crate::peerstate::{Peerstate, PeerstateVerifiedStatus};
 use crate::{chat, stock_str};
 
+/// Contact ID, including reserved IDs.
+///
+/// Some contact IDs are reserved to identify special contacts.  This
+/// type can represent both the special as well as normal contacts.
+#[derive(
+    Debug, Copy, Clone, Default, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize,
+)]
+pub struct ContactId(u32);
+
+impl ContactId {
+    /// Creates a new [`ContactId`].
+    pub const fn new(id: u32) -> ContactId {
+        ContactId(id)
+    }
+
+    /// Bad evil escape hatch, do not use.
+    pub const fn to_u32(&self) -> u32 {
+        self.0
+    }
+}
+
+impl fmt::Display for ContactId {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.0)
+        // TODO: Something like this
```

Comment:
Fix this TODO?

Reply:
I didn't do this on purpose because I want this PR to be completely a no-op while changing the display is a change.  So I left this todo as a followup, happy to just delete the todo if you prefer though.